x-pack/plugins/cases/public/components/all_cases/use_all_cases_state.tsx

```diff
@@ -0,0 +1,224 @@
+
+  return result;
+};
+
+const getFilterOptions = (
+  filterOptions: FilterOptions,
+  params: FilterOptions,
+  urlParams?: PartialFilterOptions,
+  localStorageFilterOptions?: PartialFilterOptions
+): FilterOptions => {
+  const severity =
+    params?.severity ??
+    urlParams?.severity ??
+    localStorageFilterOptions?.severity ??
+    DEFAULT_FILTER_OPTIONS.severity;
+  const status =
+    params?.status ??
+    urlParams?.status ??
+    localStorageFilterOptions?.status ??
+    DEFAULT_FILTER_OPTIONS.status;
+
+  return {
+    ...filterOptions,
+    ...params,
+    ...(severity && { severity }),
+    ...(status && { status }),
+  };
+};
+
+export function useAllCasesState(
+  isModalView: boolean = false,
+  initialFilterOptions?: PartialFilterOptions
+) {
+  const { appId } = useCasesContext();
+  const location = useLocation();
+  const history = useHistory();
+  const isFirstRenderRef = useRef(true);
+
+  const [queryParams, setQueryParams] = useState<QueryParams>({ ...DEFAULT_QUERY_PARAMS });
+  const [filterOptions, setFilterOptions] = useState<FilterOptions>({
+    ...DEFAULT_FILTER_OPTIONS,
+    ...initialFilterOptions,
+  });
+
+  const [localStorageQueryParams, setLocalStorageQueryParams] =
+    useLocalStorage<LocalStorageQueryParams>(getQueryParamsLocalStorageKey(appId));
+
+  const [localStorageFilterOptions, setLocalStorageFilterOptions] =
+    useLocalStorage<PartialFilterOptions>(getFilterOptionsLocalStorageKey(appId));
+
+  const persistAndUpdateQueryParams = useCallback(
+    (params) => {
+      if (isModalView) {
+        setQueryParams((prevParams) => ({ ...prevParams, ...params }));
+        return;
+      }
+
+      const parsedUrlParams: ParsedUrlQueryParams = parse(location.search);
+      const urlParams: PartialQueryParams = parseUrlQueryParams(parsedUrlParams);
+      const newQueryParams: QueryParams = getQueryParams(
+        params,
+        queryParams,
+        urlParams,
+        localStorageQueryParams
+      );
+      const newLocalStorageQueryParams = {
+        perPage: newQueryParams.perPage,
+        sortField: newQueryParams.sortField,
+        sortOrder: newQueryParams.sortOrder,
+      };
+      setLocalStorageQueryParams(newLocalStorageQueryParams);
+      setQueryParams(newQueryParams);
+    },
+    [isModalView, location.search, queryParams, localStorageQueryParams, setLocalStorageQueryParams]
+  );
+
+  const persistAndUpdateFilterOptions = useCallback(
+    (params) => {
+      if (isModalView) {
+        setFilterOptions((prevParams) => ({ ...prevParams, ...params }));
+        return;
+      }
+
+      const newFilterOptions: FilterOptions = getFilterOptions(
+        filterOptions,
+        params,
+        parse(location.search),
+        localStorageFilterOptions
+      );
+
+      const newPersistedFilterOptions: PartialFilterOptions = {
+        ...(newFilterOptions.severity && { severity: newFilterOptions.severity }),
```

Comment:
I believe `severity` and `status` will always be defined so I don't think we need the check here.